app/src/main/java/com/willowtree/vocable/customviews/VocableButton.kt

```diff
@@ -36,7 +40,21 @@ open class VocableButton @JvmOverloads constructor(
     private val backgroundScope = CoroutineScope(Dispatchers.IO)
     private val uiScope = CoroutineScope(Dispatchers.Main)
 
+    private val sharedPrefs: VocableSharedPreferences by inject()
+    protected var dwellTime: Long
+
+    private val sharedPrefsListener =
```

Comment:
Would this be better provided as a singleton in Koin? Right now it'll instantiate a listener for every button when they could probably share that functionality

Reply:
What would be the best approach for this? Would the dwell time be provided by Koin and something we update with a global listener? Or would there be a custom class that overrides the base shared prefs listener that we provide with Koin? And that custom listener has something like a `getDwellTime` method.

Reply:
Another option could be to make this an `object` so you're not creating a new instance for every button, so just change this to
```
object : SharedPreferences.OnSharedPreferenceChangeListener { _, key ->
            when (key) {
                VocableSharedPreferences.DWELL_TIME -> {
                    dwellTime = sharedPrefs.getDwellTime()
                }
            }
        }
```